cockpit/src/subscriptions-client.js

```diff
@@ -106,19 +137,23 @@ function getSubscriptionDetails() {
     getDetailsRequested = false;
     gettingDetails = true;
     productsService.wait(() => {
-        productsService.ListInstalledProducts('', {}, userLang) // FIXME: use proxy settings
-            .then(result => {
-                client.subscriptionStatus.products = parseProducts(result);
-            })
-            .catch(ex => {
-                client.subscriptionStatus.error = ex;
-            })
-            .then(() => {
-                gettingDetails = false;
-                if (getDetailsRequested)
-                    getSubscriptionDetails();
-                needRender();
-            });
+        safeDBusCall(productsService, 'ListInstalledProducts', callProductsService);
+    });
+}
+
+function callProductsService() {
```

Comment:
Why do you create special function for this? Is this reason any unit testing in the future? IMHO you needn't to create special function, because you can do it in this way:

```javascript
safeDBusCall(productsService, 'ListInstalledProducts', () => {
    productsService.ListInstalledProducts('', {}, userLang) // FIXME: use proxy settings
    // ...
});
```

Reply:
Done. It is because of the Java developer in me, that isn't used to how anonymous high-order functions look :smile: